test/e2e/common/pods.go

```diff
@@ -831,4 +831,103 @@ var _ = framework.KubeDescribe("Pods", func() {
 		validatePodReadiness(false)
 
 	})
+
+	ginkgo.It("pod should be in running state after the graceful terimination delete call", func() {
+		value := strconv.Itoa(time.Now().Nanosecond())
+		pod := getPodObject(podClient, f, value)
+		selector := labels.SelectorFromSet(labels.Set(map[string]string{"time": value}))
+		options := metav1.ListOptions{LabelSelector: selector.String()}
+		listCompleted := make(chan bool, 1)
+		lw := &cache.ListWatch{
+			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+				options.LabelSelector = selector.String()
+				podList, err := podClient.List(options)
+				if err == nil {
+					select {
+					case listCompleted <- true:
+						e2elog.Logf("observed the pod list")
+						return podList, err
+					default:
+						e2elog.Logf("channel blocked")
+					}
+				}
+				return podList, err
+			},
+			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+				options.LabelSelector = selector.String()
+				return podClient.Watch(options)
+			},
+		}
+		_, _, w, _ := watchtools.NewIndexerInformerWatcher(lw, &v1.Pod{})
+		defer w.Stop()
+
+		ginkgo.By("submitting the pod to kubernetes")
+		podClient.CreateSync(pod)
+
+		// We need to wait for the pod to be running, otherwise the deletion
+		// may be carried out immediately rather than gracefully.
+		framework.ExpectNoError(f.WaitForPodRunning(pod.Name))
+
+		ginkgo.By("verifying the pod is in kubernetes")
+		pods, err := podClient.List(options)
+		framework.ExpectNoError(err, "failed to query for pods")
+		gomega.Expect(len(pods.Items)).To(gomega.Equal(1))
+		pod = &pods.Items[0]
+
+		ginkgo.By("deleting the pod")
+		err = podClient.Delete(pod.Name, metav1.NewDeleteOptions(30))
+		framework.ExpectNoError(err, "failed to delete pod")
+
+		pod, err = podClient.Get(pod.Name, metav1.GetOptions{})
+
```

Comment:
Can you check my previous above comment at line 881 which is marked as solved again?
That doesn't seem to be solved..